### PR TITLE
Localize Task Center

### PR DESCRIPTION
**Pull request by cpeel**
This localizes all the UI strings in the Task Center *except* the values for the task fields.

This also doesn't localize any of the notification emails.

Testable in [localize-tasks](https://www.pgdp.org/~cpeel/c.branch/localize-tasks/) sandbox.

### CODE REVIEW

File: tasks.php [hunk 1]
```diff
@@ -322,12 +322,12 @@
 // -----------------------------------------------------------------------------
 
 $SearchParams_choices = array(
-    'task_status'   => array(998 => 'All Tasks', 999 => 'All Open Tasks') + $tasks_status_array,
-    'task_type'     => array(999 => 'All Task Types') + $tasks_array,
-    'task_severity' => array(999 => 'All Severities') + $severity_array,
-    'task_priority' => array(999 => 'All Priorities') + $priority_array,
-    'task_assignee' => array(999 => 'All Developers') + $task_assignees_array,
-    'task_category' => array(999 => 'All Categories') + $categories_array,
+    'task_status'   => array(998 => _('All Tasks'), 999 => _('All Open Tasks')) + $tasks_status_array,
```
Comment (70ray) [inline tasks.php:325]: Should we translate some of these arrays also?

**Review by chrismiceli — APPROVED** [review]

File: tasks.php (hunk 1 above)
Comment (cpeel) [reply tasks.php:325]: I intentionally didn't do that (see my MR description). My reasoning at this time is that we will need to provide translators details on what each value means to have a meaningful translation and I'm not sure the value is worth the effort.

**Review by 70ray — APPROVED** [review]

**Review by srjfoo — COMMENTED** [review]
Looks like there are several other sneaky
```return sprintf("%s by %s",``` ...
starting about line 1673 in your branch where the `"%s by %s"` should probably get the `gettext()` markup

File: tasks.php [hunk 2]
```diff
@@ -1242,25 +1240,25 @@ function TaskDetails($tid)
             if (empty($already_notified))
             {
                 echo "<input type='hidden' name='action' value='notify_me'>\n";
-                echo "<input type='submit' value='Signup for task notifications'>\n";
+                echo "<input type='submit' value='" . attr_safe(_("Signup for task notifications")) . "'>\n";
```
[the remark this answers is not in the document]
Comment (cpeel) [reply tasks.php:1243]: Good catch. I've changed this to `Sign up` here.

File: tasks.php [hunk 3]
```diff
@@ -1000,7 +1001,7 @@ function hideSpan(id) { document.getElementById(id).style.display="none"; }
     echo "<b>Show Task #</b>";
```
[the remark this answers is not in the document]
Comment (cpeel) [reply tasks.php:1001]: Yes, thanks for catching that!

File: tasks.php [hunk 4]
```diff
@@ -1079,7 +1080,7 @@ function select_and_list_tasks($sql_condition)
     $num_tasks_returned = mysqli_num_rows($sql_result);
 
     if ($num_tasks_returned == 0) {
-        echo "<p>No tasks found!</p>";
+        echo "<p>" . _("No tasks found."). "</p>";
```
[the remark this answers is not in the document]
Comment (cpeel) [reply tasks.php:1083]: Fixed!

**Comment by cpeel** [comment]
> Looks like there are several other sneaky
> `return sprintf("%s by %s",` ...
> starting about line 1673 in your branch where the `"%s by %s"` should probably get the `gettext()` markup

I decided to change `by` to an em-dash and remove the whole problem altogether. Let me know what you think after seeing it.

**Comment by srjfoo** [comment]
> I decided to change `by` to an em-dash and remove the whole problem altogether. Let me know what you think after seeing it.

😁 -- good solution. Works for me!

**Review by srjfoo — APPROVED** [review]

**Pull request merged**